bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/CarouselImpl.java

```diff
@@ -32,12 +34,40 @@
 import com.adobe.cq.wcm.core.components.sandbox.models.Carousel;
 import com.day.cq.wcm.api.components.Component;
 import com.day.cq.wcm.api.components.ComponentManager;
+import com.day.cq.wcm.api.designer.Style;
 
 @Model(adaptables = SlingHttpServletRequest.class, adapters = {Carousel.class, ComponentExporter.class}, resourceType = CarouselImpl.RESOURCE_TYPE)
 @Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
 public class CarouselImpl extends AbstractContainerImpl implements Carousel {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CarouselImpl.class);
 
-    public final static String RESOURCE_TYPE = "core/wcm/sandbox/components/carousel/v1/carousel";
+    public static final String RESOURCE_TYPE = "core/wcm/sandbox/components/carousel/v1/carousel";
+    protected static final int DEFAULT_DELAY = 5; // seconds
+
+    @ScriptVariable
+    protected Style currentStyle;
+
+    @ScriptVariable
+    protected ValueMap properties;
+
+    protected boolean auto;
+    protected Long delay;
+
+    @PostConstruct
+    protected void initModel() {
+        auto = properties.get(PN_AUTO, currentStyle.get(PN_AUTO, false));
+        delay = (long)1000 * properties.get(PN_DELAY, currentStyle.get(PN_DELAY, DEFAULT_DELAY));
```

Comment:
seconds -> milliseconds conversion

Reply:
I would rather we do the conversion where it's really needed (in the JS transform method)

Reply:
I did it this way to keep the JS api clean, it's strange to provide a timing to a JS object in seconds rather than milliseconds.

Reply:
To me the Java and JCR APIs look _dirty_ now. I do think it makes sense to keep the actual value the user entered and expose that from the Sling Model as well.

Another option would be to have a `delay` in seconds and expose a `delayMillis` property but that's ugly too.